app/src/main/java/com/example/triptracker/view/map/RecordScreen.kt

```diff
@@ -0,0 +1,406 @@
+import com.example.triptracker.navigation.getCurrentLocation
+import com.example.triptracker.view.theme.Montserrat
+import com.example.triptracker.viewmodel.RecordViewModel
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.model.CameraPosition
+import com.google.android.gms.maps.model.LatLng
+import com.google.maps.android.compose.GoogleMap
+import com.google.maps.android.compose.MapProperties
+import com.google.maps.android.compose.MapType
+import com.google.maps.android.compose.MapUiSettings
+import com.google.maps.android.compose.Polyline
+import com.google.maps.android.compose.rememberCameraPositionState
+import java.util.concurrent.TimeUnit
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+// Define color constants
+val lightDark = Color(0xFF1D2022)
+val redFox = Color(0xFFD4622B)
+val lightGray = Color(0xFFC0C7CD)
+val transparentGray = Color(0xC0C0C7CD)
+
+// Define the delay for the location update
+const val DELAY = 5000L
+
+/**
+ * Composable function for displaying the RecordScreen.
+ *
+ * @param context The application context.
+ * @param viewModel The RecordViewModel instance.
+ */
+@Composable
+fun RecordScreen(context: Context, viewModel: RecordViewModel = RecordViewModel()) {
+  // default device location is EPFL
+  var deviceLocation = LatLng(46.518831258, 6.559331096)
+
+  // Default map properties and UI settings
+  var mapProperties =
+      MapProperties(
+          mapType = MapType.NORMAL, isMyLocationEnabled = checkForLocationPermission(context))
+  var uiSettings = MapUiSettings(myLocationButtonEnabled = checkForLocationPermission(context))
+
+  // Get current device location
+  getCurrentLocation(context = context, onLocationFetched = { deviceLocation = it })
+
+  // Check for location permission
+  when (checkForLocationPermission(context = context)) {
+    true -> {
+      Map(context, viewModel, deviceLocation)
+    }
+    false -> {
+      AllowLocationPermission(
+          onPermissionGranted = {
+            mapProperties = MapProperties(mapType = MapType.NORMAL, isMyLocationEnabled = true)
+            uiSettings = MapUiSettings(myLocationButtonEnabled = true)
+          },
+          onPermissionDenied = {
+            mapProperties = MapProperties(mapType = MapType.NORMAL, isMyLocationEnabled = false)
+            uiSettings = MapUiSettings(myLocationButtonEnabled = false)
+          })
+    }
+  }
+}
+/**
+ * Composable function for displaying the map.
+ *
+ * @param context The application context.
+ * @param viewModel The RecordViewModel instance.
+ * @param startLocation The starting location.
+ */
+@Composable
+fun Map(context: Context, viewModel: RecordViewModel, startLocation: LatLng) {
+  // Mutable state for the device location and the local LatLng list
+  var deviceLocation by remember { mutableStateOf(startLocation) }
+  val localLatLngList = remember { mutableStateListOf<LatLng>() }
```

Comment:
This can be part of the view model so that later we can better implement the upload of the path to the cloud